cmd/adwatchd/integration_tests/adwatchd_service_test.go

```diff
@@ -97,15 +85,53 @@ func TestServiceStateChange(t *testing.T) {
 					return
 				}
 				require.NoError(t, err, fmt.Sprintf("%s failed but shouldn't", state))
-
-				if tc.wantStopped {
-					out := getStatus(t, app)
-					require.Contains(t, out, "stopped", "Service should be stopped")
-				}
 			}
 		})
 	}
 }
+
+func TestServiceStartWithABadDir(t *testing.T) {
+	// This should error on Windows but doesn't with systemd because of the auto-restart policy
+	if runtime.GOOS != "windows" {
+		t.Skip()
+	}
+
+	var err error
+
+	watchDir := t.TempDir()
+	configPath := generateConfig(t, -1, watchDir)
+
+	app := commands.New(commands.WithServiceName("adwatchd-test-baddir"))
+
+	installService(t, configPath, app)
+	time.Sleep(time.Second)
+
+	// Begin with a stopped state
+	changeAppArgs(t, app, "", "service", "stop")
+	err = app.Run()
+	require.NoError(t, err, "Setup: Stopping the service failed but shouldn't")
+
+	// Remove watched dir to make the service not start
+	os.RemoveAll(watchDir)
+
```

Comment:
I think I remember this, but I just want to confirm: stop is blocking, correct? This is why we don’t have a timeout here?